File: 2021-03-24/CandyTypes.java
class Solution {
    public int distributeCandies(int[] candyType) {
        //divide array for num of candies
        //get number of types
        //return the smaller of types and array length
        
        int len = candyType.length/2;
        
        Set<Integer> hash_Set = new HashSet<Integer>();
        for(int can : candyType){
            hash_Set.add(can);
        }
        int cantype = hash_Set.size();
        if(cantype < len){
            return cantype;
        }else{
            return len;
        }
    }
}

File: 2021-03-24/CandyTypes.py
class Solution(object):
    def distributeCandies(self, candyType):
        """
        :type candyType: List[int]
        :rtype: int
        """


        """divide array for num of candies
        get number of types
        return the smaller of types and array length"""
        
        leng = len(candyType)/2
        
        setC = {2}
        setC.clear()
        for can in candyType:
            setC.add(can)

        cantype = len(setC)

        if cantype < leng:
            return cantype
        else:
            return leng